tests/utils/utils.go

```diff
@@ -365,3 +368,32 @@ func calculateTxParams(ctx context.Context, wsClient ethclient.Client, fundedAdd
 
 	return gasFeeCap, gasTipCap, nonce
 }
+
+// CalculateReceiveMessageGasLimit calculates the estimated gas amount used by a single call
+// to receiveCrossChainMessage for the given message and validator bit vector. The result amount
+// depends on the required limit for the message execution, the number of validator signatures
+// included in the aggregate signature, the static gas cost defined by the precompile, and an
+// extra buffer amount defined here to ensure the call doesn't run out of gas.
+func calculateReceiveMessageGasLimit(numSigners int, executionRequiredGasLimit *big.Int) (uint64, error) {
```

Comment:
We should put this in a application (not test) utility package and export this function. I'm thinking we could have the following directory structure:
```
teleporter/
 |--- go/
 |        |--- abi-bindings/
 |        |        |--- Teleporter/
 |        |        |         |--- TeleporterMessenger/
 |        |        |         |        |---TeleporterMessenger.go
 |        |        |         |        |--- packing.go
 |        |--- teleporter-utilities/
 |        |        |--- utils.go
```